### ISSUE TITLE
with 5.0.0-beta2, clicking on a tab opens a modal

### ISSUE THREAD

**Issue by simevo** [twbs/bootstrap#33290]
- Operating system and version: Debian 11 (bullseye)
- Browser and version: chromium 88.0.4324.182 
- I am afraid it is impossible to reproduce with a reduced test case; this is happening in a Vue.js app and when I tried to factor out Vue, it disappears

I have four tabs toggled using data attributes:
```html
<ul class="nav nav-tabs nav-justified" role="tablist">
  <li class="nav-item" role="presentation">
    <button class="nav-link active" id="giorno1-tab" data-bs-toggle="tab" data-bs-target="#giorno1" type="button" role="tab" aria-controls="giorno1" aria-selected="true">Giorno 1</button>
  </li>
  <li class="nav-item" role="presentation">
    <button class="nav-link" id="giorno2-tab" data-bs-toggle="tab" data-bs-target="#giorno2" type="button" role="tab" aria-controls="giorno2" aria-selected="false">Giorno 2</button>
  </li>
  <li class="nav-item" role="presentation">
    <button class="nav-link" id="giorno3-tab" data-bs-toggle="tab" data-bs-target="#giorno3" type="button" role="tab" aria-controls="giorno3" aria-selected="false">Giorno 3</button>
  </li>
  <li class="nav-item" role="presentation">
    <button class="nav-link" id="giorno4-tab" data-bs-toggle="tab" data-bs-target="#giorno4" type="button" role="tab" aria-controls="giorno4" aria-selected="false">Giorno 4</button>
  </li>
</ul>
<div class="tab-content p-3" id="myTabContent">
  <div class="tab-pane fade show active" id="giorno1" role="tabpanel" aria-labelledby="giorno1-tab">
    ...
  </div>
  <div class="tab-pane fade" id="giorno2" role="tabpanel" aria-labelledby="giorno2-tab">
    ...
  </div>
  <div class="tab-pane fade" id="giorno3" role="tabpanel" aria-labelledby="giorno3-tab">
  </div>
    ...
  <div class="tab-pane fade" id="giorno4" role="tabpanel" aria-labelledby="giorno4-tab">
</div>
```

and a modal:
```html
<div class="modal" id="iconModal" tabindex="-1" aria-labelledby="iconModalLabel" aria-hidden="true">
  <div class="modal-dialog modal-lg">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title" id="iconModalLabel">Cambia l'icona per {{ selectedVenues }}</h5>
        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
      </div>
      <div class="modal-body">
        <div class="row" role="group" aria-label="Icons">
          ...
        </div>
      </div>
    </div>
  </div>
</div>
```

in preparation for toggling the modal programmatically I create the modal object:
```js
iconModal = new Modal(document.getElementById('iconModal'))
```

after this command is executed (but I have not yet invoked programmatically `iconModal.show()`) , clicking on any tab opens the modal before showing the matching tabbed content

more precisely:
- if I click first on the second tab, the modal opens and from then on the modal is always opened whenever I click again on the second tab (and not on the other tabs)
- but if refresh then I click first on the third tab, the modal opens from there, and from then on it is always opened whenever I click again on the third tab

I set out to debug `webpack:///./node_modules/bootstrap/dist/js/bootstrap.esm.js` with a breakpoint inside the Data Api implementation for the tab class:
```js
EventHandler.on(document, EVENT_CLICK_DATA_API$6, SELECTOR_DATA_TOGGLE$4, function (event) {
  event.preventDefault();
  var data = Data.getData(this, DATA_KEY$9) || new Tab(this);
  data.show();
});
```
to find the `data` matching the tab that was clicked, it calls `getData` which in turn calls `mapData.get`

IMHO here `storeData` looks weird; for example if I click first on the third tag (`giorno3`) then `storeData` looks like this:

```
1: Modal {_element: div#iconModal.modal, _config: {…}, _dialog: div.modal-dialog.modal-lg, _backdrop: null, _isShown: false, …}
2: Tab {_element: button#giorno2-tab.nav-link.active}
3: Tab {_element: button#giorno1-tab.nav-link}
4: Tab {_element: button#giorno4-tab.nav-link}
```

I would have expected to have an item set to `Tab {_element: button#giornog-tab.nav-link}`

what could pollute bootstrap's internal data structures like that ? can you suggest some techniques to pinpoint the problem here ? thanks

**Comment by mdo** [comment]
Without a live example, not sure what we can do. /cc @twbs/js-review for any ideas

**Comment by XhmikosR** [comment]
Please try to add a live example on CodePen. It might be a bug, though.

**Comment by simevo** [comment]
I have successfully stripped out the backend, most of the Vue components and stuff while still reproducing the issue. Only, I can not fit the code on CodePen so I have uploaded it here: https://github.com/simevo/b5-tab-modal

- Start the app with `yarnpkg serve` then open http://localhost:8080/.
- Now click on the 2nd, 3rd or 4th tab, it displays the matching tabbed content but also the modal opens (which it shouldn't).
- If for example you click first on the 2nd tab, the modal opens and from then on the modal is always opened whenever you click again on the 2nd tab (and not on the other tabs).
- But if you refresh & then click first on the 4th tab, the modal opens from there, and from then on it is always opened whenever you click again on the 4th tab.

Animated GIF:
![GIFrecord_2021-03-16_180438](https://user-images.githubusercontent.com/8254819/111350328-49629500-8682-11eb-9038-b425f62bf3d6.gif)
